capten/agent/internal/api/plugin_crossplane_provider_apis.go

```diff
@@ -116,6 +132,21 @@ func (a *Agent) UpdateCrossplanProvider(ctx context.Context, request *captenplug
 
 	a.log.Infof("Update Crossplane Provider %s, %s request recieved", request.Id, request.ProviderName)
 
+	project, err := a.as.GetCrossplanProviderByCloudType(request.CloudType)
+	if err != nil {
+		a.log.Infof("failed to get crossplane provider", err)
+		return &captenpluginspb.UpdateCrossplanProviderResponse{
+			Status:        captenpluginspb.StatusCode_INTERNAL_ERROR,
+			StatusMessage: "failed to get crossplane provider for " + request.CloudType,
+		}, nil
+	}
+	if project != nil {
+		return &captenpluginspb.UpdateCrossplanProviderResponse{
+			Status:        captenpluginspb.StatusCode_NOT_FOUND,
+			StatusMessage: "Crossplane provider is already available",
+		}, nil
+	}
```

Comment:
update would be different from add, right.
update expects already provider available. 

this implementation will never allows update.

in update case need to check project id is same being updated.